crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -520,7 +523,17 @@ where
     /// Register Debug Namespace
     pub fn register_debug(&mut self) -> &mut Self {
         let eth_api = self.eth_api();
-        self.modules.insert(RethRpcModule::Debug, DebugApi::new(eth_api).into_rpc().into());
+        self.modules.insert(
+            RethRpcModule::Debug,
+            DebugApi::new(
+                eth_api,
+                TracingCallGuard::new(Arc::new(Semaphore::new(
+                    self.config.eth.max_tracing_call_in_parallel,
```

Comment:
I'd like to move this a field of the registry, like:

https://github.com/paradigmxyz/reth/blob/8ac2b84c016ca52052c9d3999ee4e89c1245b3df/crates/rpc/rpc-builder/src/lib.rs#L424-L424

and an initializer function because we want the same instance for the TraceApi and DebugApi